src/land/madis_snow2ioda.py

```diff
@@ -0,0 +1,165 @@
+        self.outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self.varAttrs = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        self._read()
+
+    # Open input file and read relevant info
+    def _read(self):
+        # set up variable names for IODA
+        for iodavar in ['totalSnowDepth']:
+            self.varDict[iodavar]['valKey'] = iodavar, iconv.OvalName()
+            self.varDict[iodavar]['errKey'] = iodavar, iconv.OerrName()
+            self.varDict[iodavar]['qcKey'] = iodavar, iconv.OqcName()
+            self.varAttrs[iodavar, iconv.OvalName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OerrName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OqcName()]['coordinates'] = 'longitude latitude'
+            self.varAttrs[iodavar, iconv.OvalName()]['units'] = 'mm'
+            self.varAttrs[iodavar, iconv.OerrName()]['units'] = 'mm'
+
+        # open input file name
+        ncd = nc.Dataset(self.filename, 'r')
+        vals = ncd.variables['snowDepth'][:]
+        _FillValue = ncd.variables['snowDepth'].getncattr('_FillValue')
+
+        lats = ncd.variables['latitude'][:]
+        lons = ncd.variables['longitude'][:]
+        elvs = ncd.variables['elevation'][:]
+        qflg = ncd.variables['snowDepthQCR'][:]
+        obst = ncd.variables['observationTime'][:]
+        sids = nc.chartostring(ncd.variables['stationId'][:, :])
+
+        sites = np.empty_like(vals, dtype=object)
+        for i in range(len(vals[:])):
+            sites[i] = sids[i]
+        times = np.empty_like(vals, dtype=np.int64)
+        times = obst
+
+        vals = vals.astype('float32')
+        lats = lats.astype('float32')
+        lons = lons.astype('float32')
+        elvs = elvs.astype('float32')
+        errs = 0.0*vals
+        errs = errs.astype('float32')
+        qflg = qflg.astype('int32')
+
+        # use maskout options
+        if self.mask:
+            with np.errstate(invalid='ignore'):
+                mask = (vals >= 0.0) & (vals < 2000.0)
+            vals = vals[mask]
+            errs = errs[mask]
+            qflg = qflg[mask]
+            lons = lons[mask]
+            lats = lats[mask]
+            elvs = elvs[mask]
+            sites = sites[mask]
+            times = times[mask]
+
+        for i in range(len(vals)):
+            if vals[i] >= 0.0:
+                errs[i] = 40.0
+            else:
+                errs[i] = _FillValue
+
+        # add metadata variables
+        self.outdata[('dateTime', 'MetaData')] = times
+        self.outdata[('stationIdentification', 'MetaData')] = sites
+        self.outdata[('latitude', 'MetaData')] = lats
+        self.outdata[('longitude', 'MetaData')] = lons
+        self.outdata[('stationElevation', 'MetaData')] = elvs
+        self.varAttrs[('stationElevation', 'MetaData')]['units'] = 'm'
+        # self.varAttrs[('dateTime', 'MetaData')]['units'] = iso8601_string
```

Comment:
@srherbener I get a horrible crash when I try to have this line present,, I can't figure it out as have something almost identical in the ghcn_snod2ioda.py